blockmanager.go

```diff
@@ -1879,6 +1879,9 @@ func (b *blockManager) fetchFilterFromAllPeers(
 				// and wait for the next response.
 				filterResponses[sp.Addr()] = gcsFilter
 
+				// close(peerQuit) // this peer responded. don't ask again
+				log.Debugf("got gcs filter for %v from %v", blockHash, sp.Addr())
```

Comment:
I would remove this debug log entirely if it looks like the `close` was indeed required.  The debug log is added to help understand when this function is called and if it indeed makes repeated request to the same peer incorrectly after getting an initial response.

Reply:
Maybe I remove this blockManager change entirely since I am uncertain.  I was hoping that you or @ellemouton would have context on the `detectBadPeer` > `fetchFilterFromAllPeers` call path that could shed some light on this.  But at the end of the day, it's really just the `peer` fix that I was after.